storage/mdadm/include/include.sh

```diff
@@ -0,0 +1,619 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# Author: Chnaghui Zhong <czhong@redhat.com>
+
+rpm -q mdadm || yum install -y mdadm
+JOURNAL_SUPPORT=0
+info=`mdadm --create --help | grep -o "write-journal"`
+if [ "$info" = "write-journal" ]; then
+        JOURNAL_SUPPORT=1
+fi
+
+#Install fio from upstream
+function install_fio() {
+
+	git_url=git://git.kernel.org/pub/scm/linux/kernel/git/axboe/fio.git
+
+	yum install libaio-devel zlib-devel -y
+	git clone $git_url
+	echo "INFO: Installing Fio"
+	cd fio  &&./configure && make && make install
+	which fio
+	if [ $? -ne 0 ]; then
+		echo "FAIL: Fio not succesffully installed"
+		exit 1
+	fi
+
+	echo "INFO: Fio succesfully installed"
+}
+
+#Install dt
+function install_dt () { 
+
+	wget http://www.scsifaq.org/RMiller_Tools/ftp/dt/dt-source.tar.gz
+	tar xvf dt-source.tar.gz
+	cd dt.d-WIP/
+	cp -p Makefile.linux Makefile
+	make
+	cp dt /usr/bin
+}
+
+#----------------------------------------------------------------------------#
+# MD_Create_RAID ()
+# Usage:
+#   Create md raid.
+# Parameter:
+# 	$level				# like 0, 1, 3, 5, 10, 50                       
+# 	$dev_list			# like 'sda sdb sdc sdd'
+# 	$raid_dev_num		# like 3                       
+# 	$spar_dev_num		# like 2                                    
+# 	$chunk				# like 64                                     
+# Returns:
+#   Return code:
+#       0 on success
+#       1 if something went wrong.
+#   Return string:
+#       RETURN_STR		# $md_raid like '/dev/md0'
+#		MD_DEVS			# $raid_dev_list like '/dev/sda /dev/sdb'
+#----------------------------------------------------------------------------#
+
+function MD_Create_RAID (){
+	EX_USAGE=64 # Bad arg format
+	if [ $# -lt 3 ]; then
+		echo 'Usage: MD_Create_RAID $level $dev_list $raid_dev_num \
+		[$spar_dev_num] [$chunk]'
+		exit "${EX_USAGE}"
+	fi
+	# variable definitions
+	RETURN_STR=''
+	MD_DEVS=''
+	local level=$1
+	local dev_list=$2
+	local raid_dev_num=$3
+	local bitmap=$4
+	local spar_dev_num=${5:-0}
+	local chunk=${6:-512}
+	local bitmap_chunksize=${7:-64M}
+	local dev_num=0
+	local raid_dev=''
+	local spar_dev=''
+	local md_raid=""
+	local mtdata=${8:-1.2}
+	local ret=0
+	# start to create
+	echo "INFO: Executing MD_Create_RAID() to create raid $level"
+	# check if the given disks are more the needed
+	for i in $dev_list; do
+		dev_num=$((dev_num+1))
+	done
+	if [ $dev_num -lt $(($raid_dev_num+$spar_dev_num)) ]; then
+		echo "FAIL: Required devices are more than given."
+		exit 1
```

Comment:
Same here.

Actually also everywhere below with nonzero exits. This is just a test include so it all depends on the interpretation of the exit code in the test itself